Python/Include/Scalar.hpp

```diff
@@ -0,0 +1,173 @@
+#ifndef ROVER_PYTHON_SCALAR_HPP
+#define ROVER_PYTHON_SCALAR_HPP
+#include <type_traits>
+#include <dlib/algs.h>
+#include <pybind11/pybind11.h>
+
+namespace Rover {
+
+  //! PythonScalar forward declaration.
+  template<typename T>
+  class PythonScalar;
+
+  //! Pointer type for PythonScalar convertible to both T* and
+  //! PythonScalar<T>*.
+  /*
+    \tparam T The type of the scalar.
+  */
+  template<typename T>
+  class PythonScalarPointer {
+    public:
+
+      //! Constructs the object using a raw PythonScalar<T> pointer.
+      PythonScalarPointer(PythonScalar<T>* ptr);
```

Comment:
Should this constructor be ```explicit``` or do you really intend to support implicit conversions?

Reply:
I think I should even make it `private`, and the class should befriend `PythonScalar<T>`. The pointer is explicitly created in `PythonScalar<T>::operator &`, and is not supposed to be visible to an outside user.